pkg/health/aggregation_policy.go

```diff
@@ -16,26 +16,40 @@
 
 package health
 
+import "github.com/InVisionApp/go-health"
+
 // DefaultAggregationPolicy aggregates the healths by constructing a new Health based on the given
 // where the overall health status is negative if one of the healths is negative and positive if all are positive
 type DefaultAggregationPolicy struct {
 }
 
 // Apply aggregates the given healths
-func (*DefaultAggregationPolicy) Apply(healths map[string]*Health) *Health {
+func (*DefaultAggregationPolicy) Apply(healths map[string]health.State, failureTreshold int64) *Health {
 	if len(healths) == 0 {
 		return New().WithDetail("error", "no health indicators registered").Unknown()
 	}
 	overallStatus := StatusUp
 	for _, health := range healths {
-		if health.Status == StatusDown {
+		if health.Status == "failed" && health.ContiguousFailures > failureTreshold {
 			overallStatus = StatusDown
 			break
 		}
 	}
 	details := make(map[string]interface{})
 	for k, v := range healths {
-		details[k] = v
+		details[k] = ConvertStatus(v.Status)
 	}
 	return New().WithStatus(overallStatus).WithDetails(details)
 }
+
+// ConvertStatus converts go-health status to Status
+func ConvertStatus(status string) Status {
```

Comment:
I can't decide if it would be better to change our statuses from UP and Down to OK and Failed to avoid this.